lib/spack/spack/spec.py

```diff
@@ -970,7 +970,115 @@ def __init__(self, spec, name, query_parameters):
 @key_ordering
 class Spec(object):
 
-    def __init__(self, spec_like, *dep_like, **kwargs):
+    @staticmethod
+    def from_literal(spec_dict, unique=True):
+        """Builds a Spec from a dictionary containing the spec literal.
+
+        The dictionary must have a single top level key, representing the root,
+        and as many secondary level keys as needed in the spec. The definition
+        is recursive.
+
+        Args:
+            spec_dict (dict): the dictionary containing the spec literal
+            unique (bool): if True the same key appearing at different levels
```

Comment:
I would call this argument `normal`, as this is effectively what `normalize` does (removes specs with the same name).